.github/workflows/upload_plugin.yml

```
@@ -0,0 +1,84 @@
+name: Upload rpk connect plugin to S3
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      # All runs triggered by tag will really push to S3.
+      # Take care when adding more patterns here.
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+      - 'v[0-9]+.[0-9]+.[0-9]+-rc[0-9]+'
+  pull_request:
+    # Keep CI snappy for unrelated PRs
+    paths:
+      - 'resources/plugin_uploader/**'
+      - '.github/workflows/upload_plugin.yml'
+      - '.github/actions/upload_managed_plugin/**'
+      - '.goreleaser.yml'
+  workflow_dispatch: {}
+
+env:
+  # Do dry run in most cases, UNLESS the triggering event was a "tag".
+  DRY_RUN: ${{ ! github.ref_type == 'tag' }}
+
+jobs:
+  upload_rpk_connect_plugin:
+    # Let's make this fast by using a beefy runner.
+    runs-on: ubuntu-latest-32
```

```suggestion
    if: ${{ github.event.pull_request.head.repo.full_name == 'redpanda-data/connect' }}
    runs-on: ubuntu-latest-32
```
skip this job if the PR is from a forked repo or else you will get an error on configuring aws credentials because it is trying to access secrets

---

I just added a check on the repo name.   Won't check for PR-ness here, that is handled in the trigger filters.